fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/MetricUnionCompositeTableDefinition.java

```diff
@@ -93,7 +99,12 @@ public PhysicalTable build(ResourceDictionaries dictionaries, DataSourceMetadata
     private Set<PhysicalTable> getPhysicalTables(ResourceDictionaries resourceDictionaries) {
         PhysicalTableDictionary physicalTableDictionary = resourceDictionaries.getPhysicalDictionary();
         return dependentTableNames.stream()
-                .map(name -> physicalTableDictionary.getOrDefault(name, null))
+                .peek(name -> {
```

Comment:
At least mention that the table not found in physical table dictionary {dependent table name} is a table that we need to build the current table {current table name}.